### ISSUE TITLE
tabs in change form via intermediate view

### ISSUE THREAD

**Issue by etanter** [otto-torino/django-baton#143]
Hi @abidibo,
In order to ensure that a menu voice is highlighted properly, I need to use a dedicated url that internally renders the original change form. 
The url is handled by a view that only does:
```
def foo_view(request, id):
    foo_id = A.objects.get(id=int(id)).foo_id
return FooAdmin(Foo, admin.site).change_view(request, str(foo_id))
```
The strange thing is that through this indirect route, the tabs configuration in FooAdmin is ignored. It's displayed as a flat form, with the proper fieldsets and inlines, but not the tab organization.
If instead I go to the change form via the standard url (`myapp/foo/123/`), then the tabs are displayed as expected.
Any idea why this is happening?

(The reason I'm doing that is that from an `A` object, accessible via a menu model voice, I need to display the corresponding `foo` object - simply linking to the standard url works but loses the highlighting of the "parent" voice -- those `foo` objects are not accessible on their own in the menu, they're really just additional info on `A` objects.)

Thanks!

**Comment by abidibo** [comment]
Hi @etanter ,

yes, the problem was due to the way Baton autodetect the current page type (add form, change form, changelist, ...). It performs regular expressions against the current URL, and it won't activate some functionalities (i.e. tabs) if aren't necessary for the context. So tabs are activated only when the current page is of type `add_form` or `change_form`.

I've just published Baton==2.1.4 which adds the possibility to define a custom hook function used to detect the page type, you can find the documentation here:

https://github.com/otto-torino/django-baton#page-detection

Basically, you define a function `Baton.detectPageHook` which gets the Baton default page detection function as the first argument and should return the page type.

**Comment by etanter** [comment]
Thanks @abidibo! now I have a good incentive to switch to 2.x ;-)

**Comment by etanter** [comment]
works perfectly, thanks!
